pkgs/build-support/oci-tools/default.nix

```diff
@@ -1,12 +1,18 @@
 { lib, writeText, runCommand, writeReferencesToFile }:
 
+let
+  currentSystemSplit = lib.splitString "-" builtins.currentSystem;
+  currentOS = lib.last currentSystemSplit;
+  currentArch = lib.head currentSystemSplit;
+in
```

Comment:
I am pretty sure that there's better way...

Reply:
Avoid `builtins.currentSystem`. A drop-in is `hostPlatform.system`, but you probably want something like `hostPlatform.parsed.cpu.name` and `hostPlatform.parsed.kernel.name`.

Seems to match at least for x86-64 and for linux, but I don't know what format `buildContainer` really needs.